src/data/polynomial/eval.lean

```diff
@@ -792,6 +792,9 @@ end eval
 
 section map
 
+lemma map_dvd {R S} [semiring R] [comm_semiring S] (f : R →+* S) {x y : R[X]} :
```

Comment:
note that this should probably be named `map_dvd_map`, but that already exists in the form of the iff from earlier. I can rename these in another PR if people want?

Reply:
Good idea.

Reply:
I'll do it after this, so that I don't get merge conflicts and such